site/app/libraries/FileUtils.php

```diff
@@ -73,6 +73,44 @@ public static function getAllFiles(string $dir, array $skip_files=[], bool $flat
         return $return;
     }
 
+    public static function getPathWithQueryAndTip($path, $query, $tip) {
+        // https://stackoverflow.com/questions/17160696/php-glob-scan-in-subfolders-for-a-file
+        $iter = new \RecursiveDirectoryIterator($path);
+        foreach (new \RecursiveIteratorIterator($iter) as $file){
+            if (strpos($file->getFilename() , $tip) !== false) {
+                if (preg_match("/.*" . $query . ".*" . $tip . "$/i", $file->getPathname()) > 0) {
+                    return $file->getPathname();
+                }
+            }
+        }
+        return false;
+    }
+
+    public static function getDetails($path, $query) {
```

Comment:
This is also very specific to the feature we are creating and should not go in the FileUtils namespace. It just clutters up this already fairly large file.

Reply:
We could just move this into `HomePageController.php`.